infra/network/src/main/java/com/simprints/infra/network/httpclient/DefaultOkHttpClientBuilder.kt

```diff
@@ -70,10 +70,12 @@ internal class DefaultOkHttpClientBuilder @Inject constructor(
             }
 
     private fun buildRequestIdInterceptor() = Interceptor { chain ->
-        val newRequest = chain.request().newBuilder()
-            .addHeader(REQUEST_ID_HEADER, UUID.randomUUID().toString())
-            .build()
-        return@Interceptor chain.proceed(newRequest)
+        val requestId = UUID.randomUUID().toString()
+
+        // Adding same header to both request and response to
+        // track the request across network logs and analytics events.
+        val newRequest = chain.request().newBuilder().addHeader(REQUEST_ID_HEADER, requestId).build()
+        return@Interceptor chain.proceed(newRequest).newBuilder().addHeader(REQUEST_ID_HEADER, requestId).build()
```

Comment:
Smart!

Reply:
Not really, but this works. :D IMO, this is a bit of misuse, since the client should not add the response data 🤷🏻

Reply:
Agreed, but as far as hacks go, this is probably the smallest possible change you could make to make this work 😂 Code golf winner